pkgs/by-name/vula/package.nix

```diff
@@ -3,64 +3,61 @@
   python3,
   fetchgit,
   highctidh,
-  coreutils,
-}:
-with builtins; let
-  python = python3;
+  wrapGAppsHook,
+}: let
+  inherit
+    (lib)
+    licenses
+    maintainers
+    ;
 in
-  python.pkgs.buildPythonApplication rec {
+  python3.pkgs.buildPythonApplication {
     pname = "vula";
-    version = "0.2.2023112801";
-    format = "pyproject";
+    version = "unstable-2024-05-17";
 
     src = fetchgit {
       url = "https://codeberg.org/vula/vula";
-      rev = "v${version}";
-      hash = "sha256-hBB6jKCLwgfPsgINuvGuLgihrr9zhG46V6/G0SXdCSc=";
+      rev = "b82933c2d45496afb91727e7ce3dff61ae262473";
+      hash = "sha256-DVjEg28GFmA3fOgXZ8MQ7rwfZtt6WkK1qHnyTnYbKcY=";
     };
 
+    # seems like this project can be built either using pyproject.toml or setuptools.
+    # stragely, even though `format="setuptools"`, pyproject.toml is used.
     postPatch = ''
-      substituteInPlace configs/systemd/* \
-        --replace 'ExecStart=vula' "ExecStart=$out/bin/vula"
-
-      substituteInPlace configs/dbus/* \
-        --replace 'Exec=/bin/false' "Exec=${coreutils}/bin/false"
+      rm pyproject.toml
     '';
 
-    propagatedBuildInputs = with python.pkgs;
-      [
-        setuptools
-        pyaudio
-        pyroute2
-        hkdf
-        pynacl
+    propagatedBuildInputs =
+      (with python3.pkgs; [
         click
+        cryptography
+        hkdf
         packaging
-        pyyaml
-        pystray
-        qrcode
         pillow
         pydbus
-        zeroconf
+        pynacl
+        pyroute2
+        pyyaml
+        qrcode
         schema
-        cryptography
         tkinter
-      ]
+        zeroconf
+      ])
       ++ [highctidh];
 
-    postInstall = ''
-      mkdir -p $out/{lib/systemd/system,/share/dbus-1/system-services}
-      cp configs/systemd/* $out/lib/systemd/system/
-      cp configs/dbus/* $out/share/dbus-1/system-services/
-    '';
+    nativeBuildInputs = [wrapGAppsHook];
+    nativeCheckInputs = with python3.pkgs; [pytestCheckHook];
 
-    doCheck = true;
+    pytestFlagsArray = [
+      # runs more tests
+      "--doctest-modules"
+    ];
 
-    meta = with lib; {
+    meta = {
       description = "Automatic local network encryption";
       homepage = "https://vula.link/";
       license = licenses.gpl3Only;
-      maintainers = with maintainers; [lorenzleutgeb];
+      maintainers = with maintainers; [lorenzleutgeb mightyiam stepbrobd];
       mainProgram = "vula";
     };
   }
```

Comment:
The GUI does not work. That would take some more time, we figured this PR should not be held up by that.

Reply:
Could you describe how it doesn't work?

Reply:
Decided to merge this PR with GUI not working and fixing that later. I personally hope later is immediately after merging this.